### PR TITLE
[Gatsby Docs Update] Initial version of new sidebar

### PR DESCRIPTION
**Pull request by flarnie**
**what is the change?:**
- Adds media queries and JS to make sidebar collapse/expand from a
  sticky bottom bar on small screen sizes
- This is very much a WIP! Tons of stuff is not finished or working as I'd like.

Let's either use this as a discussion starter or continue work on top of it. 🏗🙌

**why make this change?:**
Moving closer to Joe's fantastic design. :)

**test plan:**
Manual testing - 

short_sidebar_gif_9_14

**issue:**
Various items under the `Sidebar Component` on this checklist;
https://github.com/bvaughn/react/wiki/Gatsby-check-list

### CODE REVIEW

**Comment by flarnie** [comment]
Major concerns here:
- We want to make this maximally accessible, even in cases where JS could be disabled and/or folks are using screen readers or AX tech.
- It doesn't quite work yet. I just implemented a version of the small-screen mock.

**Comment by flarnie** [comment]
At this point if nobody has blockers I'm inclined to land this, so that anyone can work on top of it. Will do that tomorrow unless I get changes requested. 😁

**Comment by flarnie** [comment]
Just pushed an update the fixes the title in the bottom navbar:
screen shot 2017-09-18 at 5 29 35 pm

**Comment by bvaughn** [comment]
Picking this up for the moment.

File: www/src/components/StickyResponsiveSidebar/StickyResponsiveSidebar.js [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+import {Component, React} from 'react';
+import isItemActive from 'utils/isItemActive';
+import {Sticky} from 'react-sticky';
+import Sidebar from 'templates/components/Sidebar';
+import {colors, media} from 'theme';
+
+// TODO: memoize to save doing O(n) on the active section items + subitems every
+// time.
+function findActiveItemTitle(location, defaultActiveSection) {
+  const {items} = defaultActiveSection;
```
Comment (bvaughn) [inline www/src/components/StickyResponsiveSidebar/StickyResponsiveSidebar.js]: This line causes a build error for me:
> WebpackError: Cannot read property 'items' of null

Comment (bvaughn) [reply www/src/components/StickyResponsiveSidebar/StickyResponsiveSidebar.js]: Note that I happened to have this page open when running `yarn dev`:
http://localhost:8000/blog/2013/06/12/community-roundup.html

**Comment by bvaughn** [comment]
Seems a little busted? 😁 

loopty-scrolling

**Comment by joecritch** [comment]
@flarnie @bvaughn Hi folks, I've merged these changes into #10707. Is someone okay to close this? (Also, the above should be fixed on mine, but it's still WIP)

**Comment by Mathspy** [comment]
@joecritch I think you meant https://github.com/facebook/react/pull/10735

**Comment by bvaughn** [comment]
Done!

**Pull request closed without merging**